### PR TITLE
Ea 2505 subtitles library

### PR DESCRIPTION
**Pull request by SrdjanStevanetic**

### CODE REVIEW

File: annotation-client/src/test/resources/subtitle/minimal.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "body": {
     "type": "FullTextResource",
     "language": "it",
-    "value": " WEBVTT\nNOTE Paragraph\n00:00:00.988 --> 00:00:03.572\nDolo le ultime novità\ndella moda per signora\n...\n00:00:30.359 --> 00:00:33.608\ncon il grande finale:\nCarnevale con il popolo eschimese.\n\n00:00:34.199 --> 00:00:35.780\n <special char/> musica) <special char/> ",
-    "format": "text/vtt"
+    "value": "deuhdiwuehdiwuehdwiuehdwieuhdew",
```
Comment (gsergiu) [inline annotation-client/src/test/resources/subtitle/minimal.json:6]: this file is used is more integration tests, please revert back and create a new test file if needed

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/BaseAnnotationValidator.java [hunk 2]
```diff
@@ -501,13 +507,21 @@ protected void validateSubtitleOrCaption(Annotation webAnnotation)
...
 	}
 	// check if the body.format field has a valid value
 	boolean result = getConfiguration().getAnnotationSubtitlesFormats().contains(body.getContentType());
 	if (!result) {
 	    throw new PropertyValidationException(I18nConstants.ANNOTATION_INVALID_SUBTITLES_FORMATS,
-		    I18nConstants.ANNOTATION_INVALID_SUBTITLES_FORMATS, new String[] { body.getContentType() });
+		    I18nConstants.ANNOTATION_INVALID_SUBTITLES_FORMATS, new String[] { "body.format" });
+	}
+	// check if the body.value is valid 
+	try {
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/BaseAnnotationValidator.java:519]: please move this validation to own method, validateSubtitleContent and do not enable the execution until the spike ticket is concluded

**Review by gsergiu — CHANGES_REQUESTED** [review]

File: annotation-client/src/test/resources/subtitle/minimal.json (hunk 1 above)
Comment (SrdjanStevanetic) [reply annotation-client/src/test/resources/subtitle/minimal.json]: Done.

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/BaseAnnotationValidator.java (hunk 2 above)
Comment (SrdjanStevanetic) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/BaseAnnotationValidator.java]: Done.

**Pull request merged**